src/main/interface/msp_box.c

```diff
@@ -289,58 +291,23 @@ int packFlightModeFlags(boxBitmask_t *mspFlightModeFlags)
-           && (flightModeCopyMask & (1 << i))   // this flightmode is copy is enabled
-           && FLIGHT_MODE(1 << i)) {            // this flightmode is active
-            bitArraySet(&boxEnabledMask, flightMode_boxId_map[i]);
+        const int8_t boxid = flightMode_boxId_map[i];
+        if (boxid != -1) {    // boxId_e does exist for this FLIGHT_MODE
+            bitArraySet(&boxFlightModeMask, boxid);
+            if (FLIGHT_MODE(1 << i))  // this flightmode is active
+                bitArraySet(&boxEnabledMask, boxid);
         }
     }
 
     // enable BOXes dependent on rcMode bits, indexes are the same.
     // only subset of BOXes depend on rcMode (non-ARM or FLIGHT_MODE), use mask to select them
     // NOTE: ARM and FLIGHT modes are potentially contingent on other conditions.
     //       Therefore, they must be masked/enabled separately from simple "range conditions" (RC)
-#define BM(x) (1ULL << (x))
-    // limited to 64 BOXes now to keep code simple
-    const uint64_t rcModeCopyMask = BM(BOXANTIGRAVITY)
-                                  | BM(BOXHEADADJ)
-                                  | BM(BOXCAMSTAB)
-                                  | BM(BOXCAMTRIG)
-                                  | BM(BOXBEEPERON)
-                                  | BM(BOXLEDMAX)
-                                  | BM(BOXLEDLOW)
-                                  | BM(BOXLLIGHTS)
-                                  | BM(BOXCALIB)
-                                  | BM(BOXGOV)
-                                  | BM(BOXOSD)
-                                  | BM(BOXTELEMETRY)
-                                  | BM(BOXGTUNE)
-                                  | BM(BOXSERVO1)
-                                  | BM(BOXSERVO2)
-                                  | BM(BOXSERVO3)
-                                  | BM(BOXBLACKBOX)
-                                  | BM(BOXAIRMODE)
-                                  | BM(BOX3DDISABLE)
-                                  | BM(BOXFPVANGLEMIX)
-                                  | BM(BOXBLACKBOXERASE)
-                                  | BM(BOXCAMERA1)
-                                  | BM(BOXCAMERA2)
-                                  | BM(BOXCAMERA3)
-                                  | BM(BOXFLIPOVERAFTERCRASH)
-                                  | BM(BOXPREARM)
-                                  | BM(BOXBEEPGPSCOUNT)
-                                  | BM(BOX3DONASWITCH)
-                                  | BM(BOXVTXPITMODE);
-    STATIC_ASSERT(sizeof(rcModeCopyMask) * 8 >= CHECKBOX_ITEM_COUNT, copy_mask_too_small_for_boxes);
-    for (unsigned i = 0; i < CHECKBOX_ITEM_COUNT; i++) {
-        if ((rcModeCopyMask & BM(i))    // mode copy is enabled
-           && IS_RC_MODE_ACTIVE(i)) {    // mode is active
+    for (unsigned i = 1; i < CHECKBOX_ITEM_COUNT; i++) {    // skip BOXARM
+        if (!bitArrayGet(&boxFlightModeMask, i) && IS_RC_MODE_ACTIVE(i))
```

Comment:
IMO it would be better to skip ARM explicitly (for example by marking it in boxFlightModeMask, possibly after changing its name). Hardcoding `BOXARM = 0` may cause problems in future ...

Reply:
Good point. I didn't even realise that this was done there.

Reply:
Ah, yes. I agree. Old habit from programming 8-bit microcontrollers. I’ll fix that.